lib/src/revset.rs

```diff
@@ -2777,17 +2778,41 @@ impl PartialSymbolResolver for ChangePrefixResolver<'_> {
         repo: &dyn Repo,
         symbol: &str,
     ) -> Result<Option<CommitId>, RevsetResolutionError> {
-        if let Some(prefix) = HexPrefix::try_from_reverse_hex(symbol) {
-            match self.try_resolve(repo, &prefix)? {
+        let (change_id, offset) = if let Some((prefix, suffix)) = symbol.split_once('/') {
+            if prefix.is_empty() || suffix.is_empty() {
+                return Ok(None);
+            }
+            let Ok(offset) = suffix.parse() else {
+                return Ok(None);
+            };
+            (prefix, Some(offset))
+        } else {
+            (symbol, None)
+        };
+        let Some(prefix) = HexPrefix::try_from_reverse_hex(change_id) else {
+            return Ok(None);
+        };
+        let Some(targets) = self.try_resolve(repo, &prefix)? else {
+            return Ok(None);
+        };
+        if let Some(offset) = offset {
+            if let Some(commit_id) = targets.at_offset(offset) {
+                Ok(Some(commit_id.clone()))
+            } else {
+                Err(RevsetResolutionError::InvalidChangeOffset {
+                    symbol: change_id.to_owned(),
+                    offset,
+                })
```

Comment:
Maybe we should return `Ok(None)` here? It seems odd that `present(prefix/overflow)` is an error. There might also be an extension resolver that can look up `change-id/integer` differently.